src/store/VideoContext.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useReducer, useContext } from 'react';
+import videoReducer from './VideoReducer';
+
+const VideoContext = React.createContext({
+  videos: [],
+  currentVideo: {},
+  favoriteVideos: [],
+});
+
+export const useVideoContext = () => useContext(VideoContext);
+
+function VideosProvider({ children }) {
+  const [state, dispatch] = useReducer(videoReducer, {
+    currentVideo: {},
+    favoriteVideos: [],
+    videos: [],
+  });
+
+  return (
+    <VideoContext.Provider value={{ state, dispatch }}>{children}</VideoContext.Provider>
```

Comment:
it would be better to expose helper functions like `setCurrentVideo` and `getFavoriteVideos` instead of `state` and `dispatch` directly.